README.md

```diff
@@ -48,61 +48,193 @@ When a job is available, the controller will create a pod to acquire and run the
 The entrypoint rewriting and ordering logic is heavily inspired by [the approach used in Tekton](https://github.com/tektoncd/pipeline/blob/933e4f667c19eaf0a18a19557f434dbabe20d063/docs/developers/README.md#entrypoint-rewriting-and-step-ordering).
 
 ## Architecture
-
 ```mermaid
 sequenceDiagram
-    participant bc as buildkite controller
-    participant gql as Buildkite GraphQL API
-    participant bapi as Buildkite API
-    participant kubernetes
-    bc->>gql: Get scheduled builds & jobs
-    gql-->>bc: {build: jobs: [{uuid: "abc"}]}
-    kubernetes->>pod: start
-    bc->>kubernetes: watch for pod completions
-    bc->>kubernetes: create pod with agent sidecar
-    kubernetes->>pod: create
-    pod->>bapi: agent accepts & starts job
-    pod->>pod: run sidecars
-    pod->>pod: agent bootstrap
-    pod->>pod: run user pods to completion
-    pod->>bapi: upload artifacts, exit code
-    pod->>pod: agent exit
-    kubernetes->>bc: pod completion event
-    bc->>kubernetes: cleanup finished pods
+    participant BC as Buildkite Controller
+    participant GQL as Buildkite GraphQL API
+    participant API as Buildkite API
+    participant K8S as Kubernetes
+    participant Pod
+
+    BC->>GQL: 1. Fetch scheduled builds & jobs
+    GQL-->>BC: 2. Respond with builds & jobs<br>{build: jobs: [{uuid: "abc"}]}
+    
+    BC->>K8S: 3. Create pod with agent sidecar
+    K8S->>Pod: 4. Initialize pod
+    
+    BC->>K8S: 5. Watch for pod completions
+    Pod->>API: 6. Agent accepts & starts job
+    Pod->>Pod: 7. Execute sidecars
+    Pod->>Pod: 8. Agent bootstraps environment
+    Pod->>Pod: 9. Run user-defined tasks
+    Pod->>API: 10. Upload artifacts and exit code
+    Pod->>Pod: 11. Agent exits
+    
+    K8S->>BC: 12. Notify pod completion
+    BC->>K8S: 13. Cleanup completed pods
 ```
-
 ## Installation
 
 ### Requirements
 
-- A Kubernetes cluster
+- A Kubernetes cluster (GKE, AKS, etc.)
 - An API token with the [GraphQL scope enabled](https://buildkite.com/docs/apis/graphql-api#authentication)
 - An [agent token](https://buildkite.com/docs/agent/v3/tokens)
+- A Cluster UUID (if using [Buildkite Clusters](https://buildkite.com/docs/agent/clusters))
+
+
+### Creating the K8s cluster
 
-### Deploy with Helm
+**(For simplicity, this guide uses AWS EKS, but you can use any cloud provider or on-premises Kubernetes cluster.)**
 
-The simplest way to get up and running is by deploying our [Helm](https://helm.sh) chart:
+To begin, we need to create the infrastuture that we are going to deploy our Agent into. 
```

Comment:
To avoid confusion on `Agent` can we specify `agent-stack-k8s`

Reply:
Sure! 😄